src/System.Text.Json/tests/Serialization/TestClasses.NonGenericCollections.cs

```diff
@@ -0,0 +1,354 @@
+        public void Initialize()
+        {
+            MyQueueWrapper = new QueueWrapper(new List<object> { "Hello" });
+        }
+    }
+
+    public class IEnumerableWrapper : IEnumerable
+    {
+        private readonly List<object> _list = new List<object>();
+
+        public IEnumerableWrapper() { }
+
+        public IEnumerableWrapper(List<object> items)
+        {
+            foreach (object item in items)
+            {
+                _list.Add(item);
+            }
+        }
+
+        public IEnumerator GetEnumerator()
+        {
+            return _list.GetEnumerator();
+        }
+    }
+
+    public class ICollectionWrapper : ICollection
+    {
+        private readonly List<object> _list = new List<object>();
+
+        public ICollectionWrapper() { }
+
+        public ICollectionWrapper(List<object> items)
+        {
+            foreach (object item in items)
+            {
+                _list.Add(item);
+            }
+        }
+
+        public int Count => _list.Count;
+
+        public bool IsSynchronized => ((ICollection)_list).IsSynchronized;
+
+        public object SyncRoot => ((ICollection)_list).SyncRoot;
+
+        public void CopyTo(Array array, int index)
+        {
+            ((ICollection)_list).CopyTo(array, index);
+        }
+
+        public IEnumerator GetEnumerator()
+        {
+            return _list.GetEnumerator();
+        }
+    }
+
+    public class IListWrapper : IList
+    {
+        private readonly List<object> _list = new List<object>();
+
+        public object this[int index] { get => ((IList)_list)[index]; set => ((IList)_list)[index] = value; }
+
+        public bool IsFixedSize => ((IList)_list).IsFixedSize;
+
+        public bool IsReadOnly => ((IList)_list).IsReadOnly;
+
+        public int Count => _list.Count;
+
+        public bool IsSynchronized => ((IList)_list).IsSynchronized;
+
+        public object SyncRoot => ((IList)_list).SyncRoot;
+
+        public int Add(object value)
+        {
+            return ((IList)_list).Add(value);
+        }
+
+        public void Clear()
+        {
+            _list.Clear();
+        }
+
+        public bool Contains(object value)
+        {
+            return ((IList)_list).Contains(value);
+        }
+
+        public void CopyTo(Array array, int index)
+        {
+            ((IList)_list).CopyTo(array, index);
+        }
+
+        public IEnumerator GetEnumerator()
+        {
+            return ((IList)_list).GetEnumerator();
+        }
+
+        public int IndexOf(object value)
+        {
+            return _list.IndexOf(value);
+        }
+
+        public void Insert(int index, object value)
+        {
+            _list.Insert(index, value);
+        }
+
+        public void Remove(object value)
+        {
+            _list.Remove(value);
+        }
+
+        public void RemoveAt(int index)
+        {
+            _list.RemoveAt(index);
+        }
+    }
+
+    public class IDictionaryWrapper : IDictionary
```

Comment:
Hmm would `WrapperForIDictionary` be better since by beginning with `I` it seems like it should be an interface.

Reply:
Or just leave out I.